apps/devhub/tests/test_views.py

```diff
@@ -2871,6 +2895,26 @@ def test_unlisted_addon_pass_validation(self, mock_sign_file):
         assert file_.status == amo.STATUS_LITE
         assert mock_sign_file.called
 
+    @mock.patch('devhub.views.sign_file')
+    def test_experiments_are_auto_signed(self, mock_sign_file):
```

Comment:
This isn't testing the same thing: when we upload a new file (on the frontend), there's an ajax call to the validator, which fills in the `FileUpload.validation` field. And to do that, we do need the permission (I'll try to move this permission checking code will to `parse_addon` as you advised below).

This test is testing something else: that once a file has been submitted, it'll be automatically signed (that's when you click on the "submit" button on the file upload page).